src/index.mjs

```diff
@@ -27,78 +35,162 @@ function directoryHasPackageJson(dir) {
 	return fs.readdirSync(dir).some((file) => file === 'package.json');
 }
 
+/**
+ * @param {string} pkgManager
+ * @return {string} The command
+ */
+function getExecuteCommand(pkgManager) {
+	switch (pkgManager) {
+		case 'npm':
+			return 'npx';
+		case 'bun':
+			return 'bunx';
+		case 'pnpm':
+		case 'yarn':
+			return `${pkgManager} dlx`;
+		default:
+			throw new Error(`"${pkgManager}" package manager is not supported`);
+	}
+}
+
+/**
+ * @param {string} errorMessage
+ */
+function cancelSetup(errorMessage = '') {
+	console.error(
+		stripIndent(`
+			${picocolors.red(picocolors.bold('Setup canceled!'))}
+		`),
+	);
+
+	if (errorMessage) {
+		console.error(
+			stripIndent(`${errorMessage}
+			`),
+		);
+	}
+
+	process.exit(1);
+}
+
 /**
  * @param {string} pkgManager
  */
-function getInstallCommand(pkgManager) {
-	return pkgManager === 'npm' ? 'install' : 'add';
+async function showPrompt(pkgManager) {
+	console.info(
+		stripIndent(`
+			We'll create a '${DEFAULT_CONFIG_FILE}' file containing:
+		`),
+	);
+
+	console.info(
+		picocolors.dim(
+			DEFAULT_CONFIG_CONTENT.split('\n')
+				.map((line) => `  ${line}`)
+				.join('\n'),
+		),
+	);
+
+	console.info(
+		stripIndent(`
+			Then add the related dependencies using:
+
+			  ${picocolors.dim(`${pkgManager} ${ADD_COMMAND}`)}
+		`),
+	);
+
+	let proceed;
+
+	try {
+		proceed = await confirm({
+			message: 'Continue?',
+		});
+	} catch (error) {
+		if (error instanceof Error && error.name === 'ExitPromptError') {
+			// silence
+		} else {
+			throw error;
+		}
+	}
+
+	if (!proceed) {
+		cancelSetup();
+	}
 }
 
 /**
  * @param {string} cwd
  * @param {string} pkgManager
  */
-function createConfig(cwd, pkgManager) {
+async function createConfig(cwd, pkgManager) {
 	const spinner = ora('Creating config...').start();
-	const existingConfig = getExistingConfigInDirectory();
+	const existingConfig = await getExistingConfigInDirectory();
 
 	if (existingConfig !== null) {
 		const basename = path.basename(existingConfig.filepath);
 		const failureMessage =
 			basename === 'package.json'
-				? 'The "stylelint" config in "package.json" already exists.'
-				: `The "${basename}" config already exists.`;
+				? 'A "stylelint" config in "package.json" already exists.'
+				: `A "${basename}" config already exists.`;
 
-		spinner.fail(`Failed to create config:\n${failureMessage} Remove it and then try again.`);
-		process.exit(1);
+		spinner.fail();
+		cancelSetup(`${failureMessage} Remove it and then try again.`);
 	}
 
 	if (!directoryHasPackageJson(cwd)) {
-		spinner.fail(
-			`Failed to create config:\npackage.json was not found. Run "${pkgManager} init" and then try again.`,
-		);
-		process.exit(1);
+		spinner.fail();
+		cancelSetup(`A "package.json" was not found. Run "${pkgManager} init" and then try again.`);
 	}
 
 	try {
-		fs.writeFileSync(DEFAULT_CONFIG_FILE, '{ "extends": ["stylelint-config-standard"] }');
+		fs.writeFileSync(DEFAULT_CONFIG_FILE, DEFAULT_CONFIG_CONTENT);
```

Comment:
[suggestion] How about adding EOL to the content? I guess many people would prefer it.

```suggestion
		fs.writeFileSync(DEFAULT_CONFIG_FILE, `${DEFAULT_CONFIG_CONTENT}\n`);
```